README.md

```diff
@@ -37,14 +37,13 @@ nexusPublishing {
 }
 ```
 
-**Important**. Users registered in Sonatype after [24 February 2021](https://central.sonatype.org/news/20210223_new-users-on-s01/) need to customize the following URLs:
+**Important**. Users registered in Sonatype after [24 February 2021](https://central.sonatype.org/news/20210223_new-users-on-s01/) need to the use the s01 host:
 
 ```gradle
 nexusPublishing {
     repositories {
-        sonatype {  //only for users registered in Sonatype after 24 Feb 2021
-            nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
-            snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
+        sonatype {
+            nexusHost = s01
         }
```

Comment:
How about the following?

```suggestion
        sonatype("s01.oss.sonatype.org")
```

and

```suggestion
        sonatype("s01.oss.sonatype.org") {
			...
		}
```

Reply:
`sonatype("s01.oss.sonatype.org")` is not really auto-completable :-/

I guess the sole point of `sonatype` is to have a pre-defined configuration for OSSRH, so having enums looks reasonable.

---

`sonatype("nexus.acme.com")` would look a bit weird

Reply:
I don't really feel comfortable hard-coding server names like `s01`. Has Sonatype shared any public roadmap for that server? Will there be `s02`, `s03` etc. at some point?

Reply:
So far there are only two instances.
Of course, it might sense to have a generic API that receives hostname, however, I believe, it is way more important and user-friendly to have an enum or something for `s01`.

Frankly speaking, I do not understand how `s01..` allows OSSRH to resolve the performance issue, and it is really really surprising for me they went with user-facing DNS change.

Reply:
Yet another alternative is to probe (e.g. list profiles request or something) the server and auto-detect the host among `oss.` and `s01.` if user omits the host.

Reply:
Eventually I'd like to get rid of the manual declaration of the default (which differs for users).

Sonatype suggested, that they might spin up another host when they start having issues with `s01`.

So for now there's only `s01.oss` and `oss`. I had implementations using strings before, but that could result in user error on configuration, so I choose to use an enum instead. Also the enum is extensible when new hosts spin up.

Also this approach works nicely in _Groovy_ and _Kotlin_.

Reply:
A possible shorthand/replacement could be 

    sonatype('s01')

Still relying on an `internal enum class NexusHost` for verification.

`toEnum(NexusHost::class.java, nexusHost)` works nicely.

Reply:
I still think configuring the hostname explicitly is clearer instead of just `s01`.

@szpak WDYT?

Reply:
Passing 's01' (or enum) might be somehow confusing, however, on the other hand, we originally created `sonatype()` to make the configuration straightforward for a default scenario, so there could be an option to use it with  `s01` in some similar convenient way.

Having just two Sonatype servers, we could have just `sonatype()` and `sonatypeS01()`. It will not scale, but I don't know about Sonatype plans to add another machines (assuming that `oss` has a lot of more projects that `s01` it might take some time to reach its capacity). Alternatively, I think that we could treat `sonatype('s01')` not as passing a host name, but rather a server "name", with some enum under the hood, allowing to control passed value and help to determine real server name (NOT just string concatenation "s01" + ".oss.sonatype..." accepting everything). Having `s01` easily completable in Idea (will it work also in build.gradle in Groovy?) is a plus for me.

Reply:
> Frankly speaking, I do not understand how s01.. allows OSSRH to resolve the performance issue, and it is really really surprising for me they went with user-facing DNS change.

@vlsi It's just my speculation, but I think that Sonatype doesn't have a problem with just serviing the released artifacts (which could be and is delegated to some CDN), but rather with the release process (creating a staging repository, closing it and releasing/dropping). It is handled by Nexus Pro 2 which might not be able to handle XXX simultaneous operations (even on a very large machine(s)). Having it scattered with a new independent system - for new people/projects - at least does not increase the original workload with new projects added every single day, again and again.

Reply:
I don't think we should be more concerned about expressiveness in the build script than auto completion. 

> Having just two Sonatype servers, we could have just `sonatype()` and `sonatypeS01()`.

I would be okay with that or `sonatype("s01")`. @szpak Which variant would you prefer?

Reply:
I added a commit implementing `sonatype(s01)`...